core/src/main/clojure/xtdb/sql/analyze.clj

```diff
@@ -542,20 +563,23 @@
                                   query-expression (scope-element (r/parent query-specification))
                                   named-join-columns (for [identifier (named-columns-join-columns (r/parent ag))]
                                                        {:identifier identifier})
-                                  column-references (all-column-references query-expression)]
+                                  column-references (all-column-references query-expression)
+                                  known-columns (map #(hash-map :identifier %) known-columns)]
                               (->> (for [{:keys [identifiers] column-table-id :table-id} column-references
                                          :when (= table-id column-table-id)]
                                      {:identifier (last identifiers)})
                                    (concat named-join-columns)
+                                   (concat known-columns)
                                    (distinct)))))]
         [(reduce
-          (fn [acc {:keys [identifier index]}]
-            (conj acc (cond-> (with-meta {:index (count acc)} {:table table})
-                        identifier (assoc :identifier identifier)
-                        (and index (nil? identifier)) (assoc :original-index index)
-                        correlation-name (assoc :qualified-column [correlation-name identifier]))))
-          []
-          projections)]))
+           (fn [acc {:keys [identifier index outer-name]}]
```

Comment:
aware the `reduce` was here before so dw about changing it now, but if we only need `(count acc)` we could use `map-indexed` instead. again, might seem like golfing but because `map-indexed` is less powerful than `reduce` you're telling the reader that they don't need to worry about different elements affecting each other, and that the result is 1-to-1 with the input - whereas, with `reduce`, anything could happen :slightly_smiling_face: